device/celestica/x86_64-cel_silverstone-r0/plugins/sfputil.py

```diff
@@ -524,7 +551,14 @@ def get_transceiver_change_event(self, timeout=0):
                 # initiate QSFP-DD software reset if failure counter > 2
                 if self.mod_failure[x] > 2:
                     self.mod_failure[x] = 0
-                    self._write_byte(x, self.IDENTITY_EEPROM_ADDR, -1, 26, 0x08)
+                    # perform a software reset if it's not a DAC (Passive Copper)
+                    # Note:
+                    # This is actually a hotfix for 'Eoptolink EOLD-134HG-02-M6, QSFPDD CMIS v3',
+                    # its module state could get stuck at ModuleLowPwr upon reinsertion, and a software
+                    # reset is necessary to get it recovered.
+                    buf = self._read_eeprom_devid(x, self.IDENTITY_EEPROM_ADDR, 85, 1)
+                    if (buf is not None) and (int(buf[0], 16) != 0x03):
```

Comment:
This is to address the DAC failure below:
pmon#sfputil.py: write failed: [Errno 110] Connection timed out

Some of the DAC (Passive Copper) do not support software reset, and hence such a write attempt will fail

Reply:
In case of DAC, should we even add failure counter and come here,  See line 553:
`self.mod_failure[x] += 1`

Reply:
another generic question,  it seems we are keep monitoring all the settings and do reset/init etc all the time.  It is a bit concerning if we that could be disruptive.  Thinking maybe we could only do these steps when we got the optical insert event,  and finish the bring up process once.  of course, we will do it again if insert happens again. Let me know.

Reply:
1. We'll actually do nothing in the case of DAC, so I'm not sure why we need to keep the counter advanced? Do we want to have any other automatic recovery for DAC?

2. The reason why this is not handled in the insertion event is that, the I2C read could fail and it was observed during the QA test last week. However, we could add a inited flag to make sure this is only performed once and only once, let me update the code logic

Reply:
1- that is exactly my point, we should not advance the counter and then avoid the write here,  we should eliminate the counter advance in the code earlier so it won't even be a failure case.

Reply:
copy that, and thanks, I'll update the code accordingly